guides/plugins/plugins/storefront/add-dynamic-content-via-ajax-calls.md

```diff
@@ -0,0 +1,138 @@
+The route annotation has an added `defaults={"XmlHttpRequest"=true}` to allow XmlHttpRequest and it returns a `JsonResponse` instead of a normal `Response`.
+Using a `JsonResponse` instead of a normal `Response` causes the data structures passed to it to be automatically turned into a `JSON` string.
+
+The following `services.xml` and `routes.xml` are identical to before mentioned article, but here they are for reference anyways:
+
+```xml
+<?xml version="1.0" encoding="UTF-8" ?>
+<routes xmlns="http://symfony.com/schema/routing"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://symfony.com/schema/routing
+        https://symfony.com/schema/routing/routing-1.0.xsd">
+
+    <import resource="../../Storefront/Controller/**/*Controller.php" type="annotation" />
+</routes>
+```
+
+```xml
+<?xml version="1.0" ?>
+
+<container xmlns="http://symfony.com/schema/dic/services" 
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <services>
+        <service id="SwagBasicExample\Storefront\Controller\ExampleController" public="true">
+            <call method="setContainer">
+                <argument type="service" id="service_container"/>
+            </call>
+        </service>
+    </services>
+</container>
+```
+
+## Preparing the Plugin
+
+Now we have to add a `plugin` to display the timestamp we get from our controller.
+
+Again this builds upon the [adding custom Javascript](./add-custom-javascript.md),
+so if you don't already now what storefront `plugins` are go ahead and read that.
+
+```javascript
+import HttpClient from 'src/service/http-client.service';
+import Plugin from 'src/plugin-system/plugin.class';
+
+
+export default class AjaxPlugin extends Plugin {
+    init() {
+        // initalize the HttpClient
+        this._client = new HttpClient();
+
+        // get refernces to the dom elements
+        this.button = this.el.children['ajax-button'];
+        this.textdiv = this.el.children['ajax-display'];
+
+        // register the events
+        this._registerEvents();
+    }
+
+    _registerEvents() {
+        // fetch the timestamp, when the button is clicked
+        this.button.onclick = this._fetch.bind(this);
+    }
+
+    _fetch() {
+        // make the  network request and call the `_setContent` function as a callback
+        this._client.get('/example', this._setContent.bind(this), 'application/json', true)
+    }
+
+    _setContent(data) {
+        // parse the response and set the `textdiv.innerHTML` to the timestamp
+        this.textdiv.innerHTML = JSON.parse(data).timestamp;
+    }
+}
+```
+
+
+
+## Adding the Template
+
+The only thing that is now left to do is to provide an template for the storefront plugin to hook into:
```

Comment:
```suggestion
The only thing that is now left, is to provide a template for the storefront plugin to hook into:
```

Reply:
`that is left now` instead of `that is now left`